### PR TITLE
React/workspaces table

### PR DESCRIPTION
**Pull request by kzalite**

### CODE REVIEW

File: app/ts/Workspaces/WorkspacesTable/WorkspacesTable.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            Workspaces <InlineHelp helpId="workspace" />
+          </Typography>
+        </Grid>
+      </Grid>
+      <Grid item xs={12}>
+        <Grid item xs={6}>
+          <Table size="small">
+            <TableHead>
+              <TableRow>
+                <TableCell>Title</TableCell>
```
Comment (chinook25) [inline app/ts/Workspaces/WorkspacesTable/WorkspacesTable.tsx:105]: we might want to put a width="100%" on the title header to make it look nicer

File: app/ts/Workspaces/WorkspacesTable/WorkspacesTable.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      '/#!/workspaces/' +
+      workspace.id +
+      '/problems/' +
+      workspace.defaultSubProblemId +
+      '/scenarios/' +
+      workspace.defaultScenarioId +
+      '/evidence'
+    );
+  }
+
```
Comment (chinook25) [inline app/ts/Workspaces/WorkspacesTable/WorkspacesTable.tsx:76]: to util

**Review by chinook25 — APPROVED** [review]

File: app/ts/Workspaces/WorkspacesTable/WorkspacesTable.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+import React, {useContext, useEffect, useState} from 'react';
+import CopyWorkspaceButton from './CopyWorkspaceButton/CopyWorkspaceButton';
+import DeleteWorkspaceButton from './DeleteWorkspaceButton/DeleteWorkspaceButton';
+
+export default function WorkspacesTable(): JSX.Element {
+  const {setError} = useContext(ErrorContext);
+  const [workspaces, setWorkspaces] = useState<IOldWorkspace[]>([]);
+
+  useEffect(() => {
+    Axios.get('/workspaces/')
```
Comment (DanielReid) [inline app/ts/Workspaces/WorkspacesTable/WorkspacesTable.tsx:24]: I don't see a deletion of the corresponding angular fetch anywhere

File: app/ts/Workspaces/WorkspacesTable/WorkspacesTable.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+      })
+      .catch(setError);
+  }, []);
+
+  function WorkspacesTableBody(): JSX.Element {
+    return (
+      <TableBody>
+        {_.map(
+          workspaces,
+          (workspace: IOldWorkspace, index: number): JSX.Element => {
```
Comment (DanielReid) [inline app/ts/Workspaces/WorkspacesTable/WorkspacesTable.tsx:36]: a bit big to be an anonymous inner function

File: app/ts/Workspaces/WorkspacesTable/WorkspacesTable.tsx [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    );
+  }
+
+  function deleteLocalWorkspace(id: string): void {
+    setWorkspaces(_.reject(workspaces, ['id', id]));
+  }
+
+  function getLink(workspace: IOldWorkspace): string {
+    return (
+      '/#!/workspaces/' +
```
Comment (DanielReid) [inline app/ts/Workspaces/WorkspacesTable/WorkspacesTable.tsx:67]: `template string please`

**Review by DanielReid — APPROVED** [review]
Small niggles, nice

File: app/ts/Workspaces/WorkspacesTable/WorkspacesTable.tsx (hunk 5 above)
Comment (chinook25) [reply app/ts/Workspaces/WorkspacesTable/WorkspacesTable.tsx:67]: Template string on one line was unreadable, so no, this is better in this case.

**Pull request merged**